dotnet/Vaas/src/Vaas/Vaas.cs

```diff
@@ -0,0 +1,181 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Net;
+using System.Net.WebSockets;
+using System.Security.Cryptography;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using System.Threading;
+using System.Threading.Tasks;
+using Vaas.Messages;
+using Websocket.Client;
+using Websocket.Client.Exceptions;
+
+namespace Vaas
+{
+    public class Vaas
+    {
+        private string Token { get; }
+
+        private WebsocketClient? Client { get; set; }
+
+        private string? SessionId { get; set; }
+
+        private bool Authenticated { get; set; }
+        
+        public Uri Url { get; set; } = new("wss://gateway-vaas.gdatasecurity.de");
+
+        private Dictionary<string, VerdictResponse> VerdictResponsesDict { get; } = new();
+
+        public Vaas(string token)
+        {
+            Token = token;
+        }
+
+        public void Connect()
+        {
+            Client = new WebsocketClient(Url, CreateWebsocketClient());
+            Client.ReconnectTimeout = null;
+            Client.MessageReceived.Subscribe(msg =>
+            {
+                if (msg.MessageType != WebSocketMessageType.Text) return;
+                var message = JsonSerializer.Deserialize<Message>(msg.Text);
+                switch (message?.Kind)
+                {
+                    case "AuthResponse":
+                        var authenticationResponse = JsonSerializer.Deserialize<AuthenticationResponse>(msg.Text);
+                        if (authenticationResponse is {Success: true})
+                        {
+                            Authenticated = true;
+                            SessionId = authenticationResponse.SessionId;
+                        }
+                        break;
+                        
+                    case "VerdictResponse":
+                        var options = new JsonSerializerOptions {Converters = {new JsonStringEnumConverter()}};
+                        var verdictResponse = JsonSerializer.Deserialize<VerdictResponse>(msg.Text,options);
+                        VerdictResponsesDict.Add(verdictResponse?.Guid ?? throw new InvalidOperationException(), verdictResponse);
+                        break;
+                }
+            });
+            Client.Start().GetAwaiter().GetResult();

```

Comment:
Client.Start().Wait()

Reply:
As far as I know, it's better to use "GetAwaiter" as "Wait" gives you an AggregateException in every error case, where "GetAwaiter" gives you the underlying exception. But that's more of taste/nitpick

Reply:
Converted to async.